src/tests/ftest/slurm_setup.py

```diff
@@ -14,16 +14,23 @@
 import sys
 from ClusterShell.NodeSet import NodeSet
 from util.general_utils import pcmd, run_task
+from avocado.utils.distro import detect
 
 
+distro_info = detect()
 SLURM_CONF = "/etc/slurm/slurm.conf"
-
+PACKAGE_VERSION = None
+if "suse" in distro_info.name.lower():
+    distro = "suse.lp153"
+elif "centos" in distro_info.name.lower() and distro_info.version == "7":
+    distro = "el7"
+    PACKAGE_VERSION = "18.08.8-1.{}.x86_64".format(distro)
```

Comment:
This is really rigid also.  Do we really want to have to push a whole new DAOS PR every time the slurm RPM is updated?

Honestly, the slurm packages should be a dependency of another RPM, not baked into a test like this.

Reply:
this is not a test this is the slurm_setup script that is called in ftest.

Reply:
@brianjmurrell is the concern here that the `src/tests/ftest/slurm_setup.py` script is handling the removal/installation of RPM packages that is more commonly handled by scripts like `ci/functional/required_packages.sh`?

If so, would:
- adding the `slurm`, `slurm-example-configs`, `slurm-slurmctld`, and `slurm-slurmd` packages to the `pkgs` list in `ci/functional/required_packages.sh` and
- handling the `18.08.8-1.el7.x86_64` CentOS 7 specific installation in `ci/provisioning/post_provision_config_nodes_EL_7.sh` by updating the `INST_RPMS` list
resolve this issue?

Then the `src/tests/ftest/slurm_setup.py` script could just handle configuring slurm to work based upon the installed slurm version.

Two things to consider if we were to take this approach is:

1. The functional tests would now be run with slurm always installed (but not configured) - vs only installed if running soak
2. We lose the install/uninstall functionality od slurm packages when running the `src/tests/ftest/slurm_setup.py` script manually

I guess slurm could be a dependency of one of the DAOS test RPMs, but it is only needed for soak currently.  If this were the case, then (1) from above would apply as well.

Reply:
The install/remove capabilities in the slurm_setup.py allows manual setup of nodes running soak manually so I would be opposed to removing that functionality from the script.

However, If you think the above changes to required_packages.sh/post_provision_config_el&.sh should be made; the current slurm_setup.py script can be called without the "-i" param in main.sh and would just start slurm on only the hw large nodes assigned at run time when soak is run.

Also I would not want it to be a dependency on daos test as this would be an issue when soak is run in real slurm env or pbs env. Or did you mean create a daos-client-tests-slurm package?

Reply:
> @brianjmurrell is the concern here that the `src/tests/ftest/slurm_setup.py` script is handling the removal/installation of RPM packages that is more commonly handled by scripts like `ci/functional/required_packages.sh`?

Somewhat.  The core of the concern is the rigidly specified version.  It's very specific and will need updating every time there is an update of the slurm package.  But indeed, a test really ought not to be installing packages I suspect.

That said, `ci/functional/required_packages.sh` should be avoided also.  It is really just a stop-gap to getting #3615 complete and landed.

> If so, would:
> 
> * adding the `slurm`, `slurm-example-configs`, `slurm-slurmctld`, and `slurm-slurmd` packages to the `pkgs` list in `ci/functional/required_packages.sh` and
> * handling the `18.08.8-1.el7.x86_64` CentOS 7 specific installation in `ci/provisioning/post_provision_config_nodes_EL_7.sh` by updating the `INST_RPMS` list
>   resolve this issue?

Well, we want to avoid adding more to `ci/functional/required_packages.sh` and instead utilize RPM dependencies.
 
> 1. The functional tests would now be run with slurm always installed (but not configured) - vs only installed if running soak

If we are really concerned with over-installation of dependencies, more sub-packages could be created.  The soak tests files could be put into a daos-tests-soak package, that could then depend on whatever daos-tests package supplied the functional tests and it's specific dependencies such as slurm, etc.

> 2. We lose the install/uninstall functionality od slurm packages when running the `src/tests/ftest/slurm_setup.py` script manually

By manually do you mean from a source-build, rather than an RPM build?  If so, it's a bit ugly, but a script could be provided to install the soak dependencies.  Since it is running from the source tree, it does have access to the dependency list of the daos-test-soak package in the daos.spec, so perhaps that can be leveraged so this doesn't become a dependency-sync problem.
 
> I guess slurm could be a dependency of one of the DAOS test RPMs, but it is only needed for soak currently.

Indeed, and per above, if we are concerned with the amount of (what could be deemed "optional") dependencies on the daos-tests functional testing package, sub-packages targeting specific testing could be created.  daos-tests-soak is a good example.